// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Android.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AndroidAllOf = _interopRequireDefault(require("./AndroidAllOf"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Android model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Android
 * @version 12.5.0
 */
var Android = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Android</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Android
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/AndroidAllOf
   * @param type {String} <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
   */
  function Android(type) {
    _classCallCheck(this, Android);
    _Integration["default"].initialize(this, type);
    _AndroidAllOf["default"].initialize(this);
    Android.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Android, [{
    key: "getType",
    value:
    /**
         * Returns <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
     * @param {String} type <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Your project ID from your generated private key file.
         * @return {String}
         */
  }, {
    key: "getProjectId",
    value: function getProjectId() {
      return this.projectId;
    }

    /**
     * Sets Your project ID from your generated private key file.
     * @param {String} projectId Your project ID from your generated private key file.
     */
  }, {
    key: "setProjectId",
    value: function setProjectId(projectId) {
      this['projectId'] = projectId;
    }
    /**
         * Returns Your client email from your generated private key file.
         * @return {String}
         */
  }, {
    key: "getClientEmail",
    value: function getClientEmail() {
      return this.clientEmail;
    }

    /**
     * Sets Your client email from your generated private key file.
     * @param {String} clientEmail Your client email from your generated private key file.
     */
  }, {
    key: "setClientEmail",
    value: function setClientEmail(clientEmail) {
      this['clientEmail'] = clientEmail;
    }
    /**
         * Returns Your private key from your generated private key file.
         * @return {String}
         */
  }, {
    key: "getPrivateKey",
    value: function getPrivateKey() {
      return this.privateKey;
    }

    /**
     * Sets Your private key from your generated private key file.
     * @param {String} privateKey Your private key from your generated private key file.
     */
  }, {
    key: "setPrivateKey",
    value: function setPrivateKey(privateKey) {
      this['privateKey'] = privateKey;
    }
    /**
         * Returns Your server key from the fcm console.
         * @return {String}
         */
  }, {
    key: "getServerKey",
    value: function getServerKey() {
      return this.serverKey;
    }

    /**
     * Sets Your server key from the fcm console.
     * @param {String} serverKey Your server key from the fcm console.
     */
  }, {
    key: "setServerKey",
    value: function setServerKey(serverKey) {
      this['serverKey'] = serverKey;
    }
    /**
         * Returns Your sender id from the fcm console.
         * @return {String}
         */
  }, {
    key: "getSenderId",
    value: function getSenderId() {
      return this.senderId;
    }

    /**
     * Sets Your sender id from the fcm console.
     * @param {String} senderId Your sender id from the fcm console.
     */
  }, {
    key: "setSenderId",
    value: function setSenderId(senderId) {
      this['senderId'] = senderId;
    }
    /**
         * Returns Allows users to create more than one conversation on the android integration.
         * @return {Boolean}
         */
  }, {
    key: "getCanUserCreateMoreConversations",
    value: function getCanUserCreateMoreConversations() {
      return this.canUserCreateMoreConversations;
    }

    /**
     * Sets Allows users to create more than one conversation on the android integration.
     * @param {Boolean} canUserCreateMoreConversations Allows users to create more than one conversation on the android integration.
     */
  }, {
    key: "setCanUserCreateMoreConversations",
    value: function setCanUserCreateMoreConversations(canUserCreateMoreConversations) {
      this['canUserCreateMoreConversations'] = canUserCreateMoreConversations;
    }
    /**
         * Returns Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
         * @return {Boolean}
         */
  }, {
    key: "getAttachmentsEnabled",
    value: function getAttachmentsEnabled() {
      return this.attachmentsEnabled;
    }

    /**
     * Sets Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
     * @param {Boolean} attachmentsEnabled Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
     */
  }, {
    key: "setAttachmentsEnabled",
    value: function setAttachmentsEnabled(attachmentsEnabled) {
      this['attachmentsEnabled'] = attachmentsEnabled;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {}

    /**
     * Constructs a <code>Android</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Android} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Android} The populated <code>Android</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Android();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _AndroidAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('projectId')) {
          obj['projectId'] = _ApiClient["default"].convertToType(data['projectId'], 'String');
        }
        if (data.hasOwnProperty('clientEmail')) {
          obj['clientEmail'] = _ApiClient["default"].convertToType(data['clientEmail'], 'String');
        }
        if (data.hasOwnProperty('privateKey')) {
          obj['privateKey'] = _ApiClient["default"].convertToType(data['privateKey'], 'String');
        }
        if (data.hasOwnProperty('serverKey')) {
          obj['serverKey'] = _ApiClient["default"].convertToType(data['serverKey'], 'String');
        }
        if (data.hasOwnProperty('senderId')) {
          obj['senderId'] = _ApiClient["default"].convertToType(data['senderId'], 'String');
        }
        if (data.hasOwnProperty('canUserCreateMoreConversations')) {
          obj['canUserCreateMoreConversations'] = _ApiClient["default"].convertToType(data['canUserCreateMoreConversations'], 'Boolean');
        }
        if (data.hasOwnProperty('attachmentsEnabled')) {
          obj['attachmentsEnabled'] = _ApiClient["default"].convertToType(data['attachmentsEnabled'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
 * @member {String} type
 * @default 'android'
 */
Android.prototype['type'] = 'android';

/**
 * Your project ID from your generated private key file.
 * @member {String} projectId
 */
Android.prototype['projectId'] = undefined;

/**
 * Your client email from your generated private key file.
 * @member {String} clientEmail
 */
Android.prototype['clientEmail'] = undefined;

/**
 * Your private key from your generated private key file.
 * @member {String} privateKey
 */
Android.prototype['privateKey'] = undefined;

/**
 * Your server key from the fcm console.
 * @member {String} serverKey
 */
Android.prototype['serverKey'] = undefined;

/**
 * Your sender id from the fcm console.
 * @member {String} senderId
 */
Android.prototype['senderId'] = undefined;

/**
 * Allows users to create more than one conversation on the android integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
Android.prototype['canUserCreateMoreConversations'] = undefined;

/**
 * Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
 * @member {Boolean} attachmentsEnabled
 */
Android.prototype['attachmentsEnabled'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement AndroidAllOf interface:
/**
 * <aside class=\"notice\">Firebase Cloud Messaging has deprecated its legacy APIs for HTTP and XMPP. Legacy credentials <code>serverKey</code> and <code>senderId</code> will stop working as of June 2024 and must be replaced with OAuth 2.0 access token based credentials.</aside>  To configure an android integration, first visit the [Firebase Console](https://console.firebase.google.com/).  Generate a private key from the Service accounts tab in the settings.  Copy the `project_id`, `client_email` and `private_key` from the generated JSON file and call the create integrations endpoint with this data. 
 * @member {String} type
 * @default 'android'
 */
_AndroidAllOf["default"].prototype['type'] = 'android';
/**
 * Your project ID from your generated private key file.
 * @member {String} projectId
 */
_AndroidAllOf["default"].prototype['projectId'] = undefined;
/**
 * Your client email from your generated private key file.
 * @member {String} clientEmail
 */
_AndroidAllOf["default"].prototype['clientEmail'] = undefined;
/**
 * Your private key from your generated private key file.
 * @member {String} privateKey
 */
_AndroidAllOf["default"].prototype['privateKey'] = undefined;
/**
 * Your server key from the fcm console.
 * @member {String} serverKey
 */
_AndroidAllOf["default"].prototype['serverKey'] = undefined;
/**
 * Your sender id from the fcm console.
 * @member {String} senderId
 */
_AndroidAllOf["default"].prototype['senderId'] = undefined;
/**
 * Allows users to create more than one conversation on the android integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
_AndroidAllOf["default"].prototype['canUserCreateMoreConversations'] = undefined;
/**
 * Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
 * @member {Boolean} attachmentsEnabled
 */
_AndroidAllOf["default"].prototype['attachmentsEnabled'] = undefined;
var _default = exports["default"] = Android;
