// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AndroidUpdateAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The AndroidUpdateAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AndroidUpdateAllOf
 * @version 12.5.0
 */
var AndroidUpdateAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AndroidUpdateAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AndroidUpdateAllOf
   */
  function AndroidUpdateAllOf() {
    _classCallCheck(this, AndroidUpdateAllOf);
    AndroidUpdateAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(AndroidUpdateAllOf, [{
    key: "getProjectId",
    value:
    /**
         * Returns Your project ID from your generated private key file.
         * @return {String}
         */
    function getProjectId() {
      return this.projectId;
    }

    /**
     * Sets Your project ID from your generated private key file.
     * @param {String} projectId Your project ID from your generated private key file.
     */
  }, {
    key: "setProjectId",
    value: function setProjectId(projectId) {
      this['projectId'] = projectId;
    }
    /**
         * Returns Your client email from your generated private key file.
         * @return {String}
         */
  }, {
    key: "getClientEmail",
    value: function getClientEmail() {
      return this.clientEmail;
    }

    /**
     * Sets Your client email from your generated private key file.
     * @param {String} clientEmail Your client email from your generated private key file.
     */
  }, {
    key: "setClientEmail",
    value: function setClientEmail(clientEmail) {
      this['clientEmail'] = clientEmail;
    }
    /**
         * Returns Your private key from your generated private key file.
         * @return {String}
         */
  }, {
    key: "getPrivateKey",
    value: function getPrivateKey() {
      return this.privateKey;
    }

    /**
     * Sets Your private key from your generated private key file.
     * @param {String} privateKey Your private key from your generated private key file.
     */
  }, {
    key: "setPrivateKey",
    value: function setPrivateKey(privateKey) {
      this['privateKey'] = privateKey;
    }
    /**
         * Returns Your server key from the fcm console.
         * @return {String}
         */
  }, {
    key: "getServerKey",
    value: function getServerKey() {
      return this.serverKey;
    }

    /**
     * Sets Your server key from the fcm console.
     * @param {String} serverKey Your server key from the fcm console.
     */
  }, {
    key: "setServerKey",
    value: function setServerKey(serverKey) {
      this['serverKey'] = serverKey;
    }
    /**
         * Returns Your sender id from the fcm console.
         * @return {String}
         */
  }, {
    key: "getSenderId",
    value: function getSenderId() {
      return this.senderId;
    }

    /**
     * Sets Your sender id from the fcm console.
     * @param {String} senderId Your sender id from the fcm console.
     */
  }, {
    key: "setSenderId",
    value: function setSenderId(senderId) {
      this['senderId'] = senderId;
    }
    /**
         * Returns Allows users to create more than one conversation on the android integration.
         * @return {Boolean}
         */
  }, {
    key: "getCanUserCreateMoreConversations",
    value: function getCanUserCreateMoreConversations() {
      return this.canUserCreateMoreConversations;
    }

    /**
     * Sets Allows users to create more than one conversation on the android integration.
     * @param {Boolean} canUserCreateMoreConversations Allows users to create more than one conversation on the android integration.
     */
  }, {
    key: "setCanUserCreateMoreConversations",
    value: function setCanUserCreateMoreConversations(canUserCreateMoreConversations) {
      this['canUserCreateMoreConversations'] = canUserCreateMoreConversations;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AndroidUpdateAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AndroidUpdateAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AndroidUpdateAllOf} The populated <code>AndroidUpdateAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AndroidUpdateAllOf();
        if (data.hasOwnProperty('projectId')) {
          obj['projectId'] = _ApiClient["default"].convertToType(data['projectId'], 'String');
        }
        if (data.hasOwnProperty('clientEmail')) {
          obj['clientEmail'] = _ApiClient["default"].convertToType(data['clientEmail'], 'String');
        }
        if (data.hasOwnProperty('privateKey')) {
          obj['privateKey'] = _ApiClient["default"].convertToType(data['privateKey'], 'String');
        }
        if (data.hasOwnProperty('serverKey')) {
          obj['serverKey'] = _ApiClient["default"].convertToType(data['serverKey'], 'String');
        }
        if (data.hasOwnProperty('senderId')) {
          obj['senderId'] = _ApiClient["default"].convertToType(data['senderId'], 'String');
        }
        if (data.hasOwnProperty('canUserCreateMoreConversations')) {
          obj['canUserCreateMoreConversations'] = _ApiClient["default"].convertToType(data['canUserCreateMoreConversations'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * Your project ID from your generated private key file.
 * @member {String} projectId
 */
AndroidUpdateAllOf.prototype['projectId'] = undefined;

/**
 * Your client email from your generated private key file.
 * @member {String} clientEmail
 */
AndroidUpdateAllOf.prototype['clientEmail'] = undefined;

/**
 * Your private key from your generated private key file.
 * @member {String} privateKey
 */
AndroidUpdateAllOf.prototype['privateKey'] = undefined;

/**
 * Your server key from the fcm console.
 * @member {String} serverKey
 */
AndroidUpdateAllOf.prototype['serverKey'] = undefined;

/**
 * Your sender id from the fcm console.
 * @member {String} senderId
 */
AndroidUpdateAllOf.prototype['senderId'] = undefined;

/**
 * Allows users to create more than one conversation on the android integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
AndroidUpdateAllOf.prototype['canUserCreateMoreConversations'] = undefined;
var _default = exports["default"] = AndroidUpdateAllOf;
