// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/CustomAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Webhook = _interopRequireDefault(require("./Webhook"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The CustomAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/CustomAllOf
 * @version 12.5.0
 */
var CustomAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CustomAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/CustomAllOf
   * @param webhooks {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Webhook>} An array of webhooks associated with the custom integration.
   */
  function CustomAllOf(webhooks) {
    _classCallCheck(this, CustomAllOf);
    CustomAllOf.initialize(this, webhooks);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(CustomAllOf, [{
    key: "getType",
    value:
    /**
         * Returns To configure a custom integration you need to setup a webhook with a set of triggers and target. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a custom integration you need to setup a webhook with a set of triggers and target. 
     * @param {String} type To configure a custom integration you need to setup a webhook with a set of triggers and target. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns An array of webhooks associated with the custom integration.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Webhook>}
         */
  }, {
    key: "getWebhooks",
    value: function getWebhooks() {
      return this.webhooks;
    }

    /**
     * Sets An array of webhooks associated with the custom integration.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Webhook>} webhooks An array of webhooks associated with the custom integration.
     */
  }, {
    key: "setWebhooks",
    value: function setWebhooks(webhooks) {
      this['webhooks'] = webhooks;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, webhooks) {
      obj['webhooks'] = webhooks;
    }

    /**
     * Constructs a <code>CustomAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/CustomAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/CustomAllOf} The populated <code>CustomAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CustomAllOf();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('webhooks')) {
          obj['webhooks'] = _ApiClient["default"].convertToType(data['webhooks'], [_Webhook["default"]]);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * To configure a custom integration you need to setup a webhook with a set of triggers and target. 
 * @member {String} type
 * @default 'custom'
 */
CustomAllOf.prototype['type'] = 'custom';

/**
 * An array of webhooks associated with the custom integration.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Webhook>} webhooks
 */
CustomAllOf.prototype['webhooks'] = undefined;
var _default = exports["default"] = CustomAllOf;
