// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Field = _interopRequireDefault(require("./Field"));
var _FormMessageFieldAllOf = _interopRequireDefault(require("./FormMessageFieldAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The FormMessageField model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField
 * @version 12.5.0
 */
var FormMessageField = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FormMessageField</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Field
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageFieldAllOf
   * @param type {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField.TypeEnum} The field type.
   * @param name {String} The name of the field. Must be unique per form or formResponse.
   * @param label {String} The label of the field. What the field is displayed as on Web Messenger.
   */
  function FormMessageField(type, name, label) {
    _classCallCheck(this, FormMessageField);
    _Field["default"].initialize(this, type, name, label);
    _FormMessageFieldAllOf["default"].initialize(this);
    FormMessageField.initialize(this, type, name, label);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(FormMessageField, [{
    key: "getType",
    value:
    /**
         * Returns The field type.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField.TypeEnum}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The field type.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField.TypeEnum} type The field type.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The name of the field. Must be unique per form or formResponse.
         * @return {String}
         */
  }, {
    key: "getName",
    value: function getName() {
      return this.name;
    }

    /**
     * Sets The name of the field. Must be unique per form or formResponse.
     * @param {String} name The name of the field. Must be unique per form or formResponse.
     */
  }, {
    key: "setName",
    value: function setName(name) {
      this['name'] = name;
    }
    /**
         * Returns The label of the field. What the field is displayed as on Web Messenger.
         * @return {String}
         */
  }, {
    key: "getLabel",
    value: function getLabel() {
      return this.label;
    }

    /**
     * Sets The label of the field. What the field is displayed as on Web Messenger.
     * @param {String} label The label of the field. What the field is displayed as on Web Messenger.
     */
  }, {
    key: "setLabel",
    value: function setLabel(label) {
      this['label'] = label;
    }
    /**
         * Returns Specifies the response for a text field.
         * @return {String}
         */
  }, {
    key: "getText",
    value: function getText() {
      return this.text;
    }

    /**
     * Sets Specifies the response for a text field.
     * @param {String} text Specifies the response for a text field.
     */
  }, {
    key: "setText",
    value: function setText(text) {
      this['text'] = text;
    }
    /**
         * Returns Specifies the response for a email field.
         * @return {String}
         */
  }, {
    key: "getEmail",
    value: function getEmail() {
      return this.email;
    }

    /**
     * Sets Specifies the response for a email field.
     * @param {String} email Specifies the response for a email field.
     */
  }, {
    key: "setEmail",
    value: function setEmail(email) {
      this['email'] = email;
    }
    /**
         * Returns Array of objects representing the response for a field of type select. Form and formResponse messages only.
         * @return {Array.<Object>}
         */
  }, {
    key: "getSelect",
    value: function getSelect() {
      return this.select;
    }

    /**
     * Sets Array of objects representing the response for a field of type select. Form and formResponse messages only.
     * @param {Array.<Object>} select Array of objects representing the response for a field of type select. Form and formResponse messages only.
     */
  }, {
    key: "setSelect",
    value: function setSelect(select) {
      this['select'] = select;
    }
    /**
         * Returns Placeholder text for the field. Form message only.
         * @return {String}
         */
  }, {
    key: "getPlaceholder",
    value: function getPlaceholder() {
      return this.placeholder;
    }

    /**
     * Sets Placeholder text for the field. Form message only.
     * @param {String} placeholder Placeholder text for the field. Form message only.
     */
  }, {
    key: "setPlaceholder",
    value: function setPlaceholder(placeholder) {
      this['placeholder'] = placeholder;
    }
    /**
         * Returns The minimum allowed length for the response for a field of type text. Form message only.
         * minimum: 1
         * maximum: 256
         * @return {Number}
         */
  }, {
    key: "getMinSize",
    value: function getMinSize() {
      return this.minSize;
    }

    /**
     * Sets The minimum allowed length for the response for a field of type text. Form message only.
     * @param {Number} minSize The minimum allowed length for the response for a field of type text. Form message only.
     */
  }, {
    key: "setMinSize",
    value: function setMinSize(minSize) {
      this['minSize'] = minSize;
    }
    /**
         * Returns The maximum allowed length for the response for a field of type text. Form message only.
         * minimum: 1
         * maximum: 256
         * @return {Number}
         */
  }, {
    key: "getMaxSize",
    value: function getMaxSize() {
      return this.maxSize;
    }

    /**
     * Sets The maximum allowed length for the response for a field of type text. Form message only.
     * @param {Number} maxSize The maximum allowed length for the response for a field of type text. Form message only.
     */
  }, {
    key: "setMaxSize",
    value: function setMaxSize(maxSize) {
      this['maxSize'] = maxSize;
    }
    /**
         * Returns Array of objects representing options for a field of type select.
         * @return {Array.<Object>}
         */
  }, {
    key: "getOptions",
    value: function getOptions() {
      return this.options;
    }

    /**
     * Sets Array of objects representing options for a field of type select.
     * @param {Array.<Object>} options Array of objects representing options for a field of type select.
     */
  }, {
    key: "setOptions",
    value: function setOptions(options) {
      this['options'] = options;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, name, label) {
      obj['type'] = type;
      obj['name'] = name;
      obj['label'] = label;
    }

    /**
     * Constructs a <code>FormMessageField</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField} The populated <code>FormMessageField</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FormMessageField();
        _Field["default"].constructFromObject(data, obj);
        _FormMessageFieldAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'String');
        }
        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }
        if (data.hasOwnProperty('select')) {
          obj['select'] = _ApiClient["default"].convertToType(data['select'], [Object]);
        }
        if (data.hasOwnProperty('placeholder')) {
          obj['placeholder'] = _ApiClient["default"].convertToType(data['placeholder'], 'String');
        }
        if (data.hasOwnProperty('minSize')) {
          obj['minSize'] = _ApiClient["default"].convertToType(data['minSize'], 'Number');
        }
        if (data.hasOwnProperty('maxSize')) {
          obj['maxSize'] = _ApiClient["default"].convertToType(data['maxSize'], 'Number');
        }
        if (data.hasOwnProperty('options')) {
          obj['options'] = _ApiClient["default"].convertToType(data['options'], [Object]);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The field type.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/FormMessageField.TypeEnum} type
 */
FormMessageField.prototype['type'] = undefined;

/**
 * The name of the field. Must be unique per form or formResponse.
 * @member {String} name
 */
FormMessageField.prototype['name'] = undefined;

/**
 * The label of the field. What the field is displayed as on Web Messenger.
 * @member {String} label
 */
FormMessageField.prototype['label'] = undefined;

/**
 * Specifies the response for a text field.
 * @member {String} text
 */
FormMessageField.prototype['text'] = undefined;

/**
 * Specifies the response for a email field.
 * @member {String} email
 */
FormMessageField.prototype['email'] = undefined;

/**
 * Array of objects representing the response for a field of type select. Form and formResponse messages only.
 * @member {Array.<Object>} select
 */
FormMessageField.prototype['select'] = undefined;

/**
 * Placeholder text for the field. Form message only.
 * @member {String} placeholder
 */
FormMessageField.prototype['placeholder'] = undefined;

/**
 * The minimum allowed length for the response for a field of type text. Form message only.
 * @member {Number} minSize
 * @default 1
 */
FormMessageField.prototype['minSize'] = 1;

/**
 * The maximum allowed length for the response for a field of type text. Form message only.
 * @member {Number} maxSize
 * @default 128
 */
FormMessageField.prototype['maxSize'] = 128;

/**
 * Array of objects representing options for a field of type select.
 * @member {Array.<Object>} options
 */
FormMessageField.prototype['options'] = undefined;

// Implement Field interface:
/**
 * The field type.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Field.TypeEnum} type
 */
_Field["default"].prototype['type'] = undefined;
/**
 * The name of the field. Must be unique per form or formResponse.
 * @member {String} name
 */
_Field["default"].prototype['name'] = undefined;
/**
 * The label of the field. What the field is displayed as on Web Messenger.
 * @member {String} label
 */
_Field["default"].prototype['label'] = undefined;
/**
 * Specifies the response for a text field.
 * @member {String} text
 */
_Field["default"].prototype['text'] = undefined;
/**
 * Specifies the response for a email field.
 * @member {String} email
 */
_Field["default"].prototype['email'] = undefined;
/**
 * Array of objects representing the response for a field of type select. Form and formResponse messages only.
 * @member {Array.<Object>} select
 */
_Field["default"].prototype['select'] = undefined;
// Implement FormMessageFieldAllOf interface:
/**
 * Placeholder text for the field. Form message only.
 * @member {String} placeholder
 */
_FormMessageFieldAllOf["default"].prototype['placeholder'] = undefined;
/**
 * The minimum allowed length for the response for a field of type text. Form message only.
 * @member {Number} minSize
 * @default 1
 */
_FormMessageFieldAllOf["default"].prototype['minSize'] = 1;
/**
 * The maximum allowed length for the response for a field of type text. Form message only.
 * @member {Number} maxSize
 * @default 128
 */
_FormMessageFieldAllOf["default"].prototype['maxSize'] = 128;
/**
 * Array of objects representing options for a field of type select.
 * @member {Array.<Object>} options
 */
_FormMessageFieldAllOf["default"].prototype['options'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
FormMessageField['TypeEnum'] = {
  /**
   * value: "email"
   * @const
   */
  "email": "email",
  /**
   * value: "select"
   * @const
   */
  "select": "select",
  /**
   * value: "text"
   * @const
   */
  "text": "text"
};
var _default = exports["default"] = FormMessageField;
