// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgun.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _MatchCriteriaBase = _interopRequireDefault(require("./MatchCriteriaBase"));
var _MatchCriteriaMailgunAllOf = _interopRequireDefault(require("./MatchCriteriaMailgunAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The MatchCriteriaMailgun model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgun
 * @version 12.5.0
 */
var MatchCriteriaMailgun = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MatchCriteriaMailgun</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgun
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaBase
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgunAllOf
   * @param type {String} The channel type.
   * @param integrationId {String} The ID of the integration to link. Must match the provided type.
   * @param address {String} The user’s email address.
   */
  function MatchCriteriaMailgun(type, integrationId, address) {
    _classCallCheck(this, MatchCriteriaMailgun);
    _MatchCriteriaBase["default"].initialize(this, type, integrationId);
    _MatchCriteriaMailgunAllOf["default"].initialize(this, address);
    MatchCriteriaMailgun.initialize(this, type, integrationId, address);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(MatchCriteriaMailgun, [{
    key: "getType",
    value:
    /**
         * Returns The channel type.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The channel type.
     * @param {String} type The channel type.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The ID of the integration to link. Must match the provided type.
         * @return {String}
         */
  }, {
    key: "getIntegrationId",
    value: function getIntegrationId() {
      return this.integrationId;
    }

    /**
     * Sets The ID of the integration to link. Must match the provided type.
     * @param {String} integrationId The ID of the integration to link. Must match the provided type.
     */
  }, {
    key: "setIntegrationId",
    value: function setIntegrationId(integrationId) {
      this['integrationId'] = integrationId;
    }
    /**
         * Returns Flag indicating whether the client will become the primary for the target conversation once linking is complete.
         * @return {Boolean}
         */
  }, {
    key: "getPrimary",
    value: function getPrimary() {
      return this.primary;
    }

    /**
     * Sets Flag indicating whether the client will become the primary for the target conversation once linking is complete.
     * @param {Boolean} primary Flag indicating whether the client will become the primary for the target conversation once linking is complete.
     */
  }, {
    key: "setPrimary",
    value: function setPrimary(primary) {
      this['primary'] = primary;
    }
    /**
         * Returns The user’s email address.
         * @return {String}
         */
  }, {
    key: "getAddress",
    value: function getAddress() {
      return this.address;
    }

    /**
     * Sets The user’s email address.
     * @param {String} address The user’s email address.
     */
  }, {
    key: "setAddress",
    value: function setAddress(address) {
      this['address'] = address;
    }
    /**
         * Returns May be specified to set the subject for the outgoing email.
         * @return {String}
         */
  }, {
    key: "getSubject",
    value: function getSubject() {
      return this.subject;
    }

    /**
     * Sets May be specified to set the subject for the outgoing email.
     * @param {String} subject May be specified to set the subject for the outgoing email.
     */
  }, {
    key: "setSubject",
    value: function setSubject(subject) {
      this['subject'] = subject;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, integrationId, address) {
      obj['type'] = type;
      obj['integrationId'] = integrationId;
      obj['address'] = address;
    }

    /**
     * Constructs a <code>MatchCriteriaMailgun</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgun} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MatchCriteriaMailgun} The populated <code>MatchCriteriaMailgun</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MatchCriteriaMailgun();
        _MatchCriteriaBase["default"].constructFromObject(data, obj);
        _MatchCriteriaMailgunAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('integrationId')) {
          obj['integrationId'] = _ApiClient["default"].convertToType(data['integrationId'], 'String');
        }
        if (data.hasOwnProperty('primary')) {
          obj['primary'] = _ApiClient["default"].convertToType(data['primary'], 'Boolean');
        }
        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
        }
        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The channel type.
 * @member {String} type
 * @default 'mailgun'
 */
MatchCriteriaMailgun.prototype['type'] = 'mailgun';

/**
 * The ID of the integration to link. Must match the provided type.
 * @member {String} integrationId
 */
MatchCriteriaMailgun.prototype['integrationId'] = undefined;

/**
 * Flag indicating whether the client will become the primary for the target conversation once linking is complete.
 * @member {Boolean} primary
 * @default true
 */
MatchCriteriaMailgun.prototype['primary'] = true;

/**
 * The user’s email address.
 * @member {String} address
 */
MatchCriteriaMailgun.prototype['address'] = undefined;

/**
 * May be specified to set the subject for the outgoing email.
 * @member {String} subject
 * @default 'New message from {appName}'
 */
MatchCriteriaMailgun.prototype['subject'] = 'New message from {appName}';

// Implement MatchCriteriaBase interface:
/**
 * The channel type.
 * @member {String} type
 */
_MatchCriteriaBase["default"].prototype['type'] = undefined;
/**
 * The ID of the integration to link. Must match the provided type.
 * @member {String} integrationId
 */
_MatchCriteriaBase["default"].prototype['integrationId'] = undefined;
/**
 * Flag indicating whether the client will become the primary for the target conversation once linking is complete.
 * @member {Boolean} primary
 * @default true
 */
_MatchCriteriaBase["default"].prototype['primary'] = true;
// Implement MatchCriteriaMailgunAllOf interface:
/**
 * The channel type.
 * @member {String} type
 * @default 'mailgun'
 */
_MatchCriteriaMailgunAllOf["default"].prototype['type'] = 'mailgun';
/**
 * The user’s email address.
 * @member {String} address
 */
_MatchCriteriaMailgunAllOf["default"].prototype['address'] = undefined;
/**
 * May be specified to set the subject for the outgoing email.
 * @member {String} subject
 * @default 'New message from {appName}'
 */
_MatchCriteriaMailgunAllOf["default"].prototype['subject'] = 'New message from {appName}';
var _default = exports["default"] = MatchCriteriaMailgun;
