// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Page.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Page model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Page
 * @version 12.5.0
 */
var Page = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Page</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Page
   */
  function Page() {
    _classCallCheck(this, Page);
    Page.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Page, [{
    key: "getAfter",
    value:
    /**
         * Returns A record id. Results will only contain the records that come after the specified record.  Only one of `after` or `before` can be provided, not both. 
         * @return {String}
         */
    function getAfter() {
      return this.after;
    }

    /**
     * Sets A record id. Results will only contain the records that come after the specified record.  Only one of `after` or `before` can be provided, not both. 
     * @param {String} after A record id. Results will only contain the records that come after the specified record.  Only one of `after` or `before` can be provided, not both. 
     */
  }, {
    key: "setAfter",
    value: function setAfter(after) {
      this['after'] = after;
    }
    /**
         * Returns A record id. Results will only contain the records that come before the specified record. Only one of `after` or `before` can be provided, not both. 
         * @return {String}
         */
  }, {
    key: "getBefore",
    value: function getBefore() {
      return this.before;
    }

    /**
     * Sets A record id. Results will only contain the records that come before the specified record. Only one of `after` or `before` can be provided, not both. 
     * @param {String} before A record id. Results will only contain the records that come before the specified record. Only one of `after` or `before` can be provided, not both. 
     */
  }, {
    key: "setBefore",
    value: function setBefore(before) {
      this['before'] = before;
    }
    /**
         * Returns The number of records to return. Does not apply to the `listMessages` endpoint.
         * minimum: 1
         * maximum: 100
         * @return {Number}
         */
  }, {
    key: "getSize",
    value: function getSize() {
      return this.size;
    }

    /**
     * Sets The number of records to return. Does not apply to the `listMessages` endpoint.
     * @param {Number} size The number of records to return. Does not apply to the `listMessages` endpoint.
     */
  }, {
    key: "setSize",
    value: function setSize(size) {
      this['size'] = size;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Page</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Page} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Page} The populated <code>Page</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Page();
        if (data.hasOwnProperty('after')) {
          obj['after'] = _ApiClient["default"].convertToType(data['after'], 'String');
        }
        if (data.hasOwnProperty('before')) {
          obj['before'] = _ApiClient["default"].convertToType(data['before'], 'String');
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'Number');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * A record id. Results will only contain the records that come after the specified record.  Only one of `after` or `before` can be provided, not both. 
 * @member {String} after
 */
Page.prototype['after'] = undefined;

/**
 * A record id. Results will only contain the records that come before the specified record. Only one of `after` or `before` can be provided, not both. 
 * @member {String} before
 */
Page.prototype['before'] = undefined;

/**
 * The number of records to return. Does not apply to the `listMessages` endpoint.
 * @member {Number} size
 * @default 25
 */
Page.prototype['size'] = 25;
var _default = exports["default"] = Page;
