// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ParticipantLeaveBodyParticipantId = _interopRequireDefault(require("./ParticipantLeaveBodyParticipantId"));
var _ParticipantLeaveBodyUserExternalId = _interopRequireDefault(require("./ParticipantLeaveBodyUserExternalId"));
var _ParticipantLeaveBodyUserId = _interopRequireDefault(require("./ParticipantLeaveBodyUserId"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The ParticipantLeaveBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBody
 * @version 12.5.0
 */
var ParticipantLeaveBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantLeaveBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBody
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId
   */
  function ParticipantLeaveBody() {
    _classCallCheck(this, ParticipantLeaveBody);
    _ParticipantLeaveBodyUserId["default"].initialize(this);
    _ParticipantLeaveBodyUserExternalId["default"].initialize(this);
    _ParticipantLeaveBodyParticipantId["default"].initialize(this);
    ParticipantLeaveBody.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ParticipantLeaveBody, [{
    key: "getUserId",
    value:
    /**
         * Returns The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
    function getUserId() {
      return this.userId;
    }

    /**
     * Sets The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} userId The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */
  }, {
    key: "setUserId",
    value: function setUserId(userId) {
      this['userId'] = userId;
    }
    /**
         * Returns The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
  }, {
    key: "getUserExternalId",
    value: function getUserExternalId() {
      return this.userExternalId;
    }

    /**
     * Sets The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} userExternalId The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */
  }, {
    key: "setUserExternalId",
    value: function setUserExternalId(userExternalId) {
      this['userExternalId'] = userExternalId;
    }
    /**
         * Returns The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
  }, {
    key: "getParticipantId",
    value: function getParticipantId() {
      return this.participantId;
    }

    /**
     * Sets The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} participantId The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */
  }, {
    key: "setParticipantId",
    value: function setParticipantId(participantId) {
      this['participantId'] = participantId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ParticipantLeaveBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBody} The populated <code>ParticipantLeaveBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantLeaveBody();
        _ParticipantLeaveBodyUserId["default"].constructFromObject(data, obj);
        _ParticipantLeaveBodyUserExternalId["default"].constructFromObject(data, obj);
        _ParticipantLeaveBodyParticipantId["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
        if (data.hasOwnProperty('userExternalId')) {
          obj['userExternalId'] = _ApiClient["default"].convertToType(data['userExternalId'], 'String');
        }
        if (data.hasOwnProperty('participantId')) {
          obj['participantId'] = _ApiClient["default"].convertToType(data['participantId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userId
 */
ParticipantLeaveBody.prototype['userId'] = undefined;

/**
 * The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userExternalId
 */
ParticipantLeaveBody.prototype['userExternalId'] = undefined;

/**
 * The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} participantId
 */
ParticipantLeaveBody.prototype['participantId'] = undefined;

// Implement ParticipantLeaveBodyUserId interface:
/**
 * The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userId
 */
_ParticipantLeaveBodyUserId["default"].prototype['userId'] = undefined;
// Implement ParticipantLeaveBodyUserExternalId interface:
/**
 * The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userExternalId
 */
_ParticipantLeaveBodyUserExternalId["default"].prototype['userExternalId'] = undefined;
// Implement ParticipantLeaveBodyParticipantId interface:
/**
 * The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} participantId
 */
_ParticipantLeaveBodyParticipantId["default"].prototype['participantId'] = undefined;
var _default = exports["default"] = ParticipantLeaveBody;
