// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The QuotedMessageExternalMessageId model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId
 * @version 12.5.0
 */
var QuotedMessageExternalMessageId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QuotedMessageExternalMessageId</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId
   * @param type {String} The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
   */
  function QuotedMessageExternalMessageId(type) {
    _classCallCheck(this, QuotedMessageExternalMessageId);
    QuotedMessageExternalMessageId.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(QuotedMessageExternalMessageId, [{
    key: "getType",
    value:
    /**
         * Returns The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
     * @param {String} type The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The external message Id of the quoted message.
         * @return {String}
         */
  }, {
    key: "getExternalMessageId",
    value: function getExternalMessageId() {
      return this.externalMessageId;
    }

    /**
     * Sets The external message Id of the quoted message.
     * @param {String} externalMessageId The external message Id of the quoted message.
     */
  }, {
    key: "setExternalMessageId",
    value: function setExternalMessageId(externalMessageId) {
      this['externalMessageId'] = externalMessageId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {
      obj['type'] = type;
    }

    /**
     * Constructs a <code>QuotedMessageExternalMessageId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId} The populated <code>QuotedMessageExternalMessageId</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QuotedMessageExternalMessageId();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('externalMessageId')) {
          obj['externalMessageId'] = _ApiClient["default"].convertToType(data['externalMessageId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
 * @member {String} type
 * @default 'externalMessageId'
 */
QuotedMessageExternalMessageId.prototype['type'] = 'externalMessageId';

/**
 * The external message Id of the quoted message.
 * @member {String} externalMessageId
 */
QuotedMessageExternalMessageId.prototype['externalMessageId'] = undefined;
var _default = exports["default"] = QuotedMessageExternalMessageId;
