// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The ReferralDetails model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails
 * @version 12.5.0
 */
var ReferralDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ReferralDetails</code>.
   * Nested object containing additional information.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails
   */
  function ReferralDetails() {
    _classCallCheck(this, ReferralDetails);
    ReferralDetails.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ReferralDetails, [{
    key: "getSource",
    value:
    /**
         * Returns The source of the referral - MESSENGER_CODE, ADS etc…
         * @return {String}
         */
    function getSource() {
      return this.source;
    }

    /**
     * Sets The source of the referral - MESSENGER_CODE, ADS etc…
     * @param {String} source The source of the referral - MESSENGER_CODE, ADS etc…
     */
  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
    /**
         * Returns The type of referral, typically OPEN-THREAD.
         * @return {String}
         */
  }, {
    key: "getType",
    value: function getType() {
      return this.type;
    }

    /**
     * Sets The type of referral, typically OPEN-THREAD.
     * @param {String} type The type of referral, typically OPEN-THREAD.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns If the referral came from an ad, this field will be present with the ad’s Id.
         * @return {String}
         */
  }, {
    key: "getAdId",
    value: function getAdId() {
      return this.adId;
    }

    /**
     * Sets If the referral came from an ad, this field will be present with the ad’s Id.
     * @param {String} adId If the referral came from an ad, this field will be present with the ad’s Id.
     */
  }, {
    key: "setAdId",
    value: function setAdId(adId) {
      this['adId'] = adId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ReferralDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} The populated <code>ReferralDetails</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ReferralDetails();
        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('adId')) {
          obj['adId'] = _ApiClient["default"].convertToType(data['adId'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The source of the referral - MESSENGER_CODE, ADS etc…
 * @member {String} source
 */
ReferralDetails.prototype['source'] = undefined;

/**
 * The type of referral, typically OPEN-THREAD.
 * @member {String} type
 */
ReferralDetails.prototype['type'] = undefined;

/**
 * If the referral came from an ad, this field will be present with the ad’s Id.
 * @member {String} adId
 */
ReferralDetails.prototype['adId'] = undefined;
var _default = exports["default"] = ReferralDetails;
