// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Referral.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ReferralDetails = _interopRequireDefault(require("./ReferralDetails"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Referral model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Referral
 * @version 12.5.0
 */
var Referral = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Referral</code>.
   * Data representing a referral object when a user is referred to a conversation. See the &lt;a href&#x3D;\&quot;https://docs.smooch.io/guide/conversation-referrals/\&quot;&gt;conversation referrals&lt;/a&gt; guide for more details. 
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Referral
   */
  function Referral() {
    _classCallCheck(this, Referral);
    Referral.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Referral, [{
    key: "getCode",
    value:
    /**
         * Returns The referral’s identifier.
         * @return {String}
         */
    function getCode() {
      return this.code;
    }

    /**
     * Sets The referral’s identifier.
     * @param {String} code The referral’s identifier.
     */
  }, {
    key: "setCode",
    value: function setCode(code) {
      this['code'] = code;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails}
         */
  }, {
    key: "getDetails",
    value: function getDetails() {
      return this.details;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} details
     */
  }, {
    key: "setDetails",
    value: function setDetails(details) {
      this['details'] = details;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Referral</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Referral} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Referral} The populated <code>Referral</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Referral();
        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('details')) {
          obj['details'] = _ReferralDetails["default"].constructFromObject(data['details']);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The referral’s identifier.
 * @member {String} code
 */
Referral.prototype['code'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} details
 */
Referral.prototype['details'] = undefined;
var _default = exports["default"] = Referral;
