// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Unity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
var _UnityAllOf = _interopRequireDefault(require("./UnityAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The Unity model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Unity
 * @version 12.5.0
 */
var Unity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Unity</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Unity
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/UnityAllOf
   * @param type {String} To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
   */
  function Unity(type) {
    _classCallCheck(this, Unity);
    _Integration["default"].initialize(this, type);
    _UnityAllOf["default"].initialize(this);
    Unity.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Unity, [{
    key: "getType",
    value:
    /**
         * Returns To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
     * @param {String} type To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Allows users to create more than one conversation on the Unity integration.
         * @return {Boolean}
         */
  }, {
    key: "getCanUserCreateMoreConversations",
    value: function getCanUserCreateMoreConversations() {
      return this.canUserCreateMoreConversations;
    }

    /**
     * Sets Allows users to create more than one conversation on the Unity integration.
     * @param {Boolean} canUserCreateMoreConversations Allows users to create more than one conversation on the Unity integration.
     */
  }, {
    key: "setCanUserCreateMoreConversations",
    value: function setCanUserCreateMoreConversations(canUserCreateMoreConversations) {
      this['canUserCreateMoreConversations'] = canUserCreateMoreConversations;
    }
    /**
         * Returns Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
         * @return {Boolean}
         */
  }, {
    key: "getAttachmentsEnabled",
    value: function getAttachmentsEnabled() {
      return this.attachmentsEnabled;
    }

    /**
     * Sets Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
     * @param {Boolean} attachmentsEnabled Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
     */
  }, {
    key: "setAttachmentsEnabled",
    value: function setAttachmentsEnabled(attachmentsEnabled) {
      this['attachmentsEnabled'] = attachmentsEnabled;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {}

    /**
     * Constructs a <code>Unity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Unity} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Unity} The populated <code>Unity</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Unity();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _UnityAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('canUserCreateMoreConversations')) {
          obj['canUserCreateMoreConversations'] = _ApiClient["default"].convertToType(data['canUserCreateMoreConversations'], 'Boolean');
        }
        if (data.hasOwnProperty('attachmentsEnabled')) {
          obj['attachmentsEnabled'] = _ApiClient["default"].convertToType(data['attachmentsEnabled'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
 * @member {String} type
 * @default 'unity'
 */
Unity.prototype['type'] = 'unity';

/**
 * Allows users to create more than one conversation on the Unity integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
Unity.prototype['canUserCreateMoreConversations'] = undefined;

/**
 * Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
 * @member {Boolean} attachmentsEnabled
 */
Unity.prototype['attachmentsEnabled'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement UnityAllOf interface:
/**
 * To configure a Unity integration, create an integration with type 'unity' by calling the Create Integration endpoint. 
 * @member {String} type
 * @default 'unity'
 */
_UnityAllOf["default"].prototype['type'] = 'unity';
/**
 * Allows users to create more than one conversation on the Unity integration.
 * @member {Boolean} canUserCreateMoreConversations
 */
_UnityAllOf["default"].prototype['canUserCreateMoreConversations'] = undefined;
/**
 * Allows users to send attachments. By default, the setting is set to true. This setting can only be configured in Zendesk Admin Center. 
 * @member {Boolean} attachmentsEnabled
 */
_UnityAllOf["default"].prototype['attachmentsEnabled'] = undefined;
var _default = exports["default"] = Unity;
