// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SourceWebhook = _interopRequireDefault(require("./SourceWebhook"));
var _User = _interopRequireDefault(require("./User"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The UserUpdateEventAllOfPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload
 * @version 12.5.0
 */
var UserUpdateEventAllOfPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserUpdateEventAllOfPayload</code>.
   * The payload of the event. The contents of this object depend on the type of event.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload
   */
  function UserUpdateEventAllOfPayload() {
    _classCallCheck(this, UserUpdateEventAllOfPayload);
    UserUpdateEventAllOfPayload.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(UserUpdateEventAllOfPayload, [{
    key: "getUser",
    value:
    /**
         * Returns The updated user.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User}
         */
    function getUser() {
      return this.user;
    }

    /**
     * Sets The updated user.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user The updated user.
     */
  }, {
    key: "setUser",
    value: function setUser(user) {
      this['user'] = user;
    }
    /**
         * Returns The reason why the user was updated, if applicable. * `authentication` - An anonymous user became an [identified](https://docs.smooch.io/guide/intro-to-users/) user. * `localeDetection` - A user was updated as a result of automated locale detection on messages sent. 
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload.ReasonEnum}
         */
  }, {
    key: "getReason",
    value: function getReason() {
      return this.reason;
    }

    /**
     * Sets The reason why the user was updated, if applicable. * `authentication` - An anonymous user became an [identified](https://docs.smooch.io/guide/intro-to-users/) user. * `localeDetection` - A user was updated as a result of automated locale detection on messages sent. 
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload.ReasonEnum} reason The reason why the user was updated, if applicable. * `authentication` - An anonymous user became an [identified](https://docs.smooch.io/guide/intro-to-users/) user. * `localeDetection` - A user was updated as a result of automated locale detection on messages sent. 
     */
  }, {
    key: "setReason",
    value: function setReason(reason) {
      this['reason'] = reason;
    }
    /**
         * Returns The source of the creation.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook}
         */
  }, {
    key: "getSource",
    value: function getSource() {
      return this.source;
    }

    /**
     * Sets The source of the creation.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source The source of the creation.
     */
  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserUpdateEventAllOfPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload} The populated <code>UserUpdateEventAllOfPayload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserUpdateEventAllOfPayload();
        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], _User["default"]);
        }
        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }
        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], _SourceWebhook["default"]);
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The updated user.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user
 */
UserUpdateEventAllOfPayload.prototype['user'] = undefined;

/**
 * The reason why the user was updated, if applicable. * `authentication` - An anonymous user became an [identified](https://docs.smooch.io/guide/intro-to-users/) user. * `localeDetection` - A user was updated as a result of automated locale detection on messages sent. 
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/UserUpdateEventAllOfPayload.ReasonEnum} reason
 */
UserUpdateEventAllOfPayload.prototype['reason'] = undefined;

/**
 * The source of the creation.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source
 */
UserUpdateEventAllOfPayload.prototype['source'] = undefined;

/**
 * Allowed values for the <code>reason</code> property.
 * @enum {String}
 * @readonly
 */
UserUpdateEventAllOfPayload['ReasonEnum'] = {
  /**
   * value: "authentication"
   * @const
   */
  "authentication": "authentication",
  /**
   * value: "localeDetection"
   * @const
   */
  "localeDetection": "localeDetection"
};
var _default = exports["default"] = UserUpdateEventAllOfPayload;
