// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/WebhookSubSchema.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Sunshine Conversations API
 *
 * The version of the OpenAPI document: 12.5.0
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 *
 * OpenAPI Generator version: 4.3.1
 *
 * Do not edit the class manually.
 *
 */
/**
 * The WebhookSubSchema model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/WebhookSubSchema
 * @version 12.5.0
 */
var WebhookSubSchema = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhookSubSchema</code>.
   * The webhook that generated the payload.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookSubSchema
   */
  function WebhookSubSchema() {
    _classCallCheck(this, WebhookSubSchema);
    WebhookSubSchema.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(WebhookSubSchema, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the webhook.
         * @return {String}
         */
    function getId() {
      return this.id;
    }

    /**
     * Sets The unique ID of the webhook.
     * @param {String} id The unique ID of the webhook.
     */
  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * Returns Schema version of the payload delivered to this webhook (v2).
         * @return {String}
         */
  }, {
    key: "getVersion",
    value: function getVersion() {
      return this.version;
    }

    /**
     * Sets Schema version of the payload delivered to this webhook (v2).
     * @param {String} version Schema version of the payload delivered to this webhook (v2).
     */
  }, {
    key: "setVersion",
    value: function setVersion(version) {
      this['version'] = version;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>WebhookSubSchema</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookSubSchema} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookSubSchema} The populated <code>WebhookSubSchema</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookSubSchema();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * The unique ID of the webhook.
 * @member {String} id
 */
WebhookSubSchema.prototype['id'] = undefined;

/**
 * Schema version of the payload delivered to this webhook (v2).
 * @member {String} version
 */
WebhookSubSchema.prototype['version'] = undefined;
var _default = exports["default"] = WebhookSubSchema;
